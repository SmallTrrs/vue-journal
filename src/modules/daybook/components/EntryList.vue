<template>
  <div class="entry-list-container">
      <div class="px-2 pt-2">
        <input 
         type="text"
         class="form-control"
         placeholder="Buscar Entrada"
         v-model = 'term'
        >
      </div>


     <div class="mt-2 d-flex flex-colum">
        <button 
         @click="$router.push({ name: 'entry' , params: {id: 'new'}})"
        class="btn btn-primary btn-sm">
            <i class="fa fa-plus-circle"></i>
            Nueva Entrada
        </button>
     </div>

      <div class="entry-scrollarea">
       <EntryComp
         v-for="entry in entriesByTerm"
         :key="entry.id"
         :entry="entry">
         
       </EntryComp>
                    
      </div>
  </div>
</template>

<script>
import { defineAsyncComponent } from 'vue'
import { mapGetters } from 'vuex'

export default {

    components: {
        EntryComp: defineAsyncComponent( ()=> import('./EntryComp.vue') )
    },
    computed:{
        ...mapGetters('journal', ['getEntriesByTerm']),
        entriesByTerm(){
            return this.getEntriesByTerm( this.term )
        }
    },
    data(){

     return {
         term: ''
     }

    }
    

}
</script>

<style lang="scss" scoped>

input{
    height: 50px;
}

.entry-list-container{
    border-right: 1px solid #2c3e50;
    height: cal(100vh - 56 );
}

.entry-scrollarea{
    height: cal( 100vh - 120);
    overflow: scroll;
}

</style>