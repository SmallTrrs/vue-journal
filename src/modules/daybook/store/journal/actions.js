import journalApi from "@/api/journalApi"


export const loadEntries = async ({ commit }) =>{

  const { data }= await   journalApi.get('/entries.json')
     

  if ( !data){

       commit('setEntries',[])
       return 
  }

  const entries = []

  for ( let id of Object.keys( data ) )
     entries.push({
         id,
         ...data[id]
     })

    commit('setEntries', entries )

}

export const createEntry = async ({ commit } , entry ) =>{

    const { date , picture , text } = entry

    const dataSave = { date, picture, text}

    const { data } = await journalApi.post('entries.json', dataSave)

    dataSave.id = data.name // data.name es el id que se manda desde el api de firebase

    commit('addEntry', dataSave)     

    return data.name // se regresa para redireccionar a la nueva entrada creada

}


export const updateEntry = async ({ commit } , entry) =>{

     const { date, picture, text } = entry

     const dataSave = { date, picture, text }

     await journalApi.put(`/entries/${entry.id}.json`, dataSave)
    
     commit('updateEntry', {...entry})

}


export const delEntry = async ({ commit } , id ) =>{

    await journalApi.delete(`/entries/${id}.json`)

    commit('deleteEntry', id )

}