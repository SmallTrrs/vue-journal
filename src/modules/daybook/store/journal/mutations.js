

export const setEntries =  ( state , entries ) =>{

    state.entries = [...state.entries, ...entries ]
    
    state.isLoading = false

}

export const createEntry =  ( /* state */ ) =>{


}
export const updateEntry =  ( /* state */ ) =>{


}