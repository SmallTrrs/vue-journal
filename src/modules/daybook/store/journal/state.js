
export default () => ({

  isLoading: true,
  entries:[
      {
        //   id:   new Date().getTime(),
          id: '1',
          date: new Date().toDateString(),
          text: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book',
          picture: null
      },
      {
        //   id: new Date().getTime() + 1000,
          id:'2',
          date: new Date().toDateString(),
          text: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book',
          picture: null
      },
      {
        //   id: new Date().getTime() + 2000,
          id:'3',
          date: new Date().toDateString(),
          text: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book',
          picture: null
      }
  ]

})

