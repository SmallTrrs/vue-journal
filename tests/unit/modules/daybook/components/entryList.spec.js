import { createStore } from 'vuex'
import { shallowMount } from '@vue/test-utils'

import journal from '@/modules/daybook/store/journal'
import  EntryList  from '@/modules/daybook/components/EntryList'
//import { getEntriesByTerm } from '@/modules/daybook/store/journal/getters'
import { journalState } from '../../../mock-data/test-journal-state'

describe('Pruebas en entryList', ()=>{


// PRIMERA FORMA DE HACERLO
    //    const journalMockModule = {
//      namespaced: true,
//      getters:{
//         getEntriesByTerm
//      }, 
//      state: ()=>({
//         isLoading: false,
//         entries: journalState.entries
//      })

//    }

//    const store = createStore({

//       modules:{
//         journal: { ...journalMockModule}
//       }
//    })

const createVuexStore = ( initialState ) => 
    createStore({
        modules: {
            journal:{
                ...journal,
                state: { ...initialState }
            }
        }
    })


  const store =  createVuexStore( journalState )

  const mockRouter = {
    push: jest.fn()
  }

  let wrapper

  beforeEach( ()=>{
      jest.clearAllMocks()

      wrapper = shallowMount(EntryList, {
        global: {
          mocks: {
               $router: mockRouter
          },
          plugins:[ store ]
        }
     })

  })

  

   test('Debe de llamar de getEntriesByTerm  sin termino y mostrar 3 entradas', ()=>{

       expect( wrapper.findAll('entry-comp-stub').length).toBe(3)
       expect( wrapper.html()).toMatchSnapshot()

   })

   test('Debe de llamar al getEntryByTerm y filtrar las entradas',async ()=>{

     const input = wrapper.find('input')

      await  input.setValue('Segunda')

      expect( wrapper.findAll('entry-comp-stub').length).toBe(1)
   })

   test('El boton de nuevo debe de redireccionar a /new',async ()=>{

     wrapper.find('button').trigger('click')
     expect(mockRouter.push ).toHaveBeenCalledWith({ name: 'entry', params: {id: 'new'}})

   })


})