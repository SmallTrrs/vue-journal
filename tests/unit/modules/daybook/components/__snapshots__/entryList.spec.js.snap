// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en entryList Debe de llamar de getEntriesByTerm  sin termino y mostrar 3 entradas 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar Entrada"></div>
  <div class="mt-2 d-flex flex-colum"><button class="btn btn-primary btn-sm"><i class="fa fa-plus-circle"></i> Nueva Entrada </button></div>
  <div class="entry-scrollarea">
    <entry-comp-stub entry="[object Object]"></entry-comp-stub>
    <entry-comp-stub entry="[object Object]"></entry-comp-stub>
    <entry-comp-stub entry="[object Object]"></entry-comp-stub>
  </div>
</div>
`;
